packages/grid/test/column-observer.test.js

```diff
@@ -0,0 +1,216 @@
+import sinon from 'sinon';
+import { ColumnObserver } from '../src/vaadin-grid-helpers.js';
+
+function createColumn() {
+  return document.createElement('vaadin-grid-column');
+}
+
+class Wrapper extends HTMLElement {
+  constructor() {
+    super();
+    this.attachShadow({ mode: 'open' }).innerHTML = `
+        <div id="host"></div>
+    `;
+  }
+}
+
+customElements.define('wrapper-component', Wrapper);
+
+describe('column observer', () => {
+  let wrapper;
+  let host;
+  let spy;
+  let observer;
+
+  beforeEach(async () => {
+    wrapper = fixtureSync('<wrapper-component></wrapper-component>');
+    host = wrapper.shadowRoot.querySelector('#host');
+    spy = sinon.spy();
+    observer = new ColumnObserver(host, spy);
+    await nextFrame();
+
+    // Expect the initial call
+    expect(spy.calledOnce).to.be.true;
+    spy.resetHistory();
+  });
+
+  it('should call the callback when a column is added', async () => {
+    const column = createColumn();
+    host.appendChild(column);
+    await nextFrame();
+
+    expect(spy.calledOnce).to.be.true;
+
+    const [addedColumns, removedColumns] = spy.getCalls()[0].args;
+    expect(addedColumns).to.deep.equal([column]);
+    expect(removedColumns).to.deep.equal([]);
+  });
+
+  it('should call the callback when a column is removed', async () => {
+    const column = createColumn();
+    host.appendChild(column);
+    await nextFrame();
+    host.removeChild(column);
+    await nextFrame();
+
+    expect(spy.calledTwice).to.be.true;
+
+    const [addedColumns, removedColumns] = spy.getCalls()[1].args;
+    expect(addedColumns).to.deep.equal([]);
+    expect(removedColumns).to.deep.equal([column]);
+  });
+
+  it('should call the callback synchronously on flush', () => {
+    const column = createColumn();
+    host.appendChild(column);
+    observer.flush();
+
+    expect(spy.calledOnce).to.be.true;
+  });
+
+  it('should not call the callback when a non-column is added', () => {
+    host.appendChild(document.createElement('div'));
+    observer.flush();
+
+    expect(spy).not.to.have.been.called;
+  });
+
+  it('should call the callback when a column order is changed', async () => {
+    const column = createColumn();
+    host.appendChild(column);
+    const column2 = createColumn();
+    host.appendChild(column2);
+    observer.flush();
+    spy.resetHistory();
+
+    // Reorder columns
+    host.appendChild(column);
+    await nextFrame();
+
+    expect(spy.calledOnce).to.be.true;
+
+    const [addedColumns, removedColumns] = spy.getCalls()[0].args;
+    expect(addedColumns).to.deep.equal([]);
+    expect(removedColumns).to.deep.equal([]);
+  });
+
+  it('should stop observing when disconnected', async () => {
+    observer.disconnect();
+    const column = createColumn();
+    host.appendChild(column);
+    await nextFrame();
+
+    expect(spy).not.to.have.been.called;
+  });
+
+  it('should call the callback initially when disconnected', async () => {
```

Comment:
The title says "should call" but the callback isn't called. Maybe "should not call"?

Reply:
Fixed the title